lighthouse-core/gather/gather-runner.js

```diff
@@ -427,6 +429,41 @@ class GatherRunner {
     };
   }
 
+  /**
+   * Uses the debugger protocol to fetch the manifest from within the context of
+   * the target page, reusing any credentials, emulation, etc, already established
+   * there.
+   *
+   * Returns the parsed manifest or null if the page had no manifest. If the manifest
+   * was unparseable as JSON, manifest.value will be undefined and manifest.warning
+   * will have the reason. See manifest-parser.js for more information.
+   *
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts['Manifest']>}
+   */
+  static async getAppManifest(passContext) {
```

Comment:
should we merge most of this into `driver.getAppManifest()`? Not the `baseArtifacts` check, of course, and probably not the parse step(?)